source/as-stream.js

```diff
@@ -50,13 +52,27 @@ module.exports = options => {
 			proxy.emit('error', new ReadError(error, options));
 		});
 
-		response.pipe(output);
-
 		if (options.throwHttpErrors && statusCode !== 304 && (statusCode < 200 || statusCode > 299)) {
 			proxy.emit('error', new HTTPError(statusCode, response.statusMessage, response.headers, options), null, response);
 			return;
 		}
 
+		finished = true;
+
+		response.pipe(output);
+
+		for (const destination of piped) {
+			if (!destination.headersSent) {
```

Comment:
Usually better with early returns to reduce nesting.

```js
if (!destination.headersSent) {
  continue;
}
```

Reply:
You misled me :P You shouldn't negate the condition :D